chaco/overlays/scatter_inspector_overlay.py

```diff
@@ -0,0 +1,165 @@
+# (C) Copyright 2006-2021 Enthought, Inc., Austin, TX
+# All rights reserved.
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in LICENSE.txt and may be redistributed only under
+# the conditions described in the aforementioned license. The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+# Thanks for using Enthought open source!
+# Major library imports
+from numpy import array, asarray
+
+# Enthought library imports
+from enable.api import ColorTrait, MarkerTrait
+from traits.api import Float, Int, Str, Trait
+from traits.observation.events import TraitChangeEvent
+
+# Local, relative imports
+from chaco.abstract_overlay import AbstractOverlay
+from chaco.scatterplot import render_markers
+
+
+class ScatterInspectorOverlay(AbstractOverlay):
+    """
+    Highlights points on a scatterplot as the mouse moves over them.
+    Can render the points in a different style, as well as display a
+    DataLabel.
+
+    Used in conjuction with ScatterInspector.
+    """
+
+    #: The style to use when a point is hovered over
+    hover_metadata_name = Str("hover")
+    hover_marker = Trait(None, None, MarkerTrait)
+    hover_marker_size = Trait(None, None, Int)
+    hover_line_width = Trait(None, None, Float)
+    hover_color = Trait(None, None, ColorTrait)
+    hover_outline_color = Trait(None, None, ColorTrait)
+
+    #: The style to use when a point has been selected by a click
+    selection_metadata_name = Str("selections")
+    selection_marker = Trait(None, None, MarkerTrait)
+    selection_marker_size = Trait(None, None, Int)
+    selection_line_width = Trait(None, None, Float)
+    selection_color = Trait(None, None, ColorTrait)
+    selection_outline_color = Trait(None, None, ColorTrait)
+
+    # For now, implement the equivalent of this Traits 3 feature manually
+    # using a series of trait change handlers (defined at the end of the
+    # class)
+    # @on_trait_change(
+    #     'component.index.metadata_changed,component.value.metadata_changed'
+    # )
+    def metadata_updated(self, event):
+        if self.component is not None:
+            self.component.request_redraw()
+
+    def overlay(self, component, gc, view_bounds=None, mode="normal"):
+        plot = self.component
+        if not plot or not plot.index or not getattr(plot, "value", True):
+            return
+
+        for inspect_type in (
+            self.hover_metadata_name,
+            self.selection_metadata_name,
+        ):
+            if inspect_type in plot.index.metadata:
+                # if hasattr(plot,"value") and \
+                #         not inspect_type in plot.value.metadata:
+                #     continue
```

Comment:
I apologize for all the line length confusion, this was a result of flake8 errors being caught in the moved files (as the moved ones weren't in tox.ini)
Rather than adding them to tox.ini I simply made the fixes now

Reply:
yep yep. understood. i was making these comments as a FTR